ecs-agent/acs/session/task_manifest_common.go

```diff
@@ -0,0 +1,73 @@
+package session
+
+import (
+	apitaskstatus "github.com/aws/amazon-ecs-agent/agent/api/task/status"
+	"github.com/aws/amazon-ecs-agent/agent/logger"
+	"github.com/aws/aws-sdk-go/aws"
+
+	"github.com/aws/amazon-ecs-agent/ecs-agent/acs/model/ecsacs"
+)
+
+// SequenceNumberAccessor stores the latest sequence number that corresponds to the
+// most recently processed task manifest message and is used to determine if a
+// message is stale.
+type SequenceNumberAccessor interface {
+	GetLatestSequenceNumber() int64
+	SetLatestSequenceNumber(sequenceNumber int64)
+}
+
+// ManifestMessageIDAccessor stores the latest message ID that corresponds to the
+// most recently processed task manifest message and is used to determine the
+// validity of a task stop verification ack message.
+type ManifestMessageIDAccessor interface {
+	GetMessageID() string
+	SetMessageID(messageID string) error
+}
+
+type TaskGetter interface {
+	GetRunningTasksOnInstance() ([]string, error)
+}
+
+type TaskStopper interface {
+	StopTask(taskARN string)
+}
+
+// CompareTasks returns a list of the running tasks that need to be stopped after comparing
+// the list of tasks running on the instance to the list of tasks on the task manifest,
+// which is the source of truth about what tasks are supposed to be running.
+func CompareTasks(
```

Comment:
This function seems like it is only needed for handling/responding to task manifest messages from ACS - IIUC it is not used for handling / responding to task manifest and task stop verification ACK messages.